src/ResourceManager/Profile/Commands.Profile/Environment/AddAzureRMEnvironment.cs

```diff
@@ -135,6 +155,57 @@ public override void ExecuteCmdlet()
                 {
                     var profileClient = new RMProfileClient(AzureRmProfileProvider.Instance.GetProfile<AzureRmProfile>());
 
+                    if (this.ParameterSetName.Equals(MetadataParameterSet, StringComparison.Ordinal))
+            		{
+                        // Simply use built-in environments if the ARM endpoint matches the ARM endpoint for a built-in environment
+                        ResourceManagerEndpoint = ARMEndpoint;
+                        bool isPublicArmEndpoint = AzureEnvironment.PublicEnvironments.Select
+                            (publicEnv => AzureRmProfileProvider.Instance.Profile.GetEnvironment(publicEnv.Key))
+                                                                   .Any(env => string.Equals(
```

Comment:
In this case, the appropriate thing would be to add the public environment using the given name.  We should refactor this to simply use AzureEnvironment.PublicEnvironments.FirstOrDefault to return the first matching environment or null, add that environment under the given name, unless it is one of the public environment names, and return it.

This would allow people to share scripts across stack and azure environments.  Your below check would just check for null in this case.